[Clojure] Adds Clojure Concept "Basic Functions" Exercise (#804)
exercism/v3: languages/clojure/exercises/concept/basic-functions/.doc/hints.md

### 1. Create a function that returns 42

You can use the `defn` macro to define a function.

The defn macro accepts the name of the function you are defining, an optional docstring, a vector of input arguments, and the expressions to be evaluated when the function is called.

### 2. Create a function that accepts an input and adds 42 to it 

You can name the argument passed into the function by writing its name within the argument vector when the function is defined. The argument can then be referenced by that name anywhere within the function.

Remember to use prefix notation with the '+' function.

> ### 3. Create a function that takes and argument and calls function defined in (2) with it

Reviewer: ```suggestion
### 3. Create a function that takes an argument and calls the function defined in (2) with it
```

> ### 3. Create a function that takes and argument and calls function defined in (2) with it

Reviewer: ```suggestion
### 3. Create a function that takes an argument and calls function defined in (2) with it
```

> Remember, you can invoke a function you defined the same way you invoke other functions- by putting its name as the first argument of a Clojure form and letting the Clojure compile evalute it!

Reviewer: ```suggestion
Remember, you can invoke a function you defined the same way you invoke other functions - by putting its name as the first argument of a Clojure form and letting the Clojure compile evaluate it!
```

Author: I think the original way with no space is proper english grammar, isn't it?

Reviewer: @JimLynchCodes you should be right, please ignore my comment then, I'm not a native speaker:-)

Author: np! Even native english speakers like me don't fully know english. 😁

> Remember, you can invoke a function you defined the same way you invoke other functions- by putting its name as the first argument of a Clojure form and letting the Clojure compile evalute it!

Reviewer 2: ```suggestion
Remember, you can invoke a function you defined the same way you invoke other functions- by putting its name as the first argument of a Clojure form and letting the Clojure compiler evaluate it!
```

> ### 1. Create a function that returns 42

Reviewer 3: I really like the hints! They are very well written.

Author: Thanks! 😊

> You can use the `defn` macro to define a function!

Reviewer 3: ```suggestion
You can use the `defn` macro to define a function.
```

> The defn macro accepts the name of the function you are defining, and optional docstring, a vector of input arguments, and the expressions to be evaluated when the function is called!

Reviewer 3: ```suggestion
The defn macro accepts the name of the function you are defining, an optional docstring, a vector of input arguments, and the expressions to be evaluated when the function is called.
```

### 2. Create a function that accepts an input and adds 42 to it 

You can name the argument passed into the function by writing its name within the argument vector when the function is defined. The argument can then be referenced by that name anywhere within the function.

Remember to use prefix notation with the '+' function. For example, in Clojure "one plus two" would be written as:

> ```clojure

Reviewer 3: In general, we want the hints to only contain code when there is no other option. I think you could lose the example here, as the "Remember to use prefix notation with the '+' function" should give the student enough context that the student can solve it themselves.

Author: removed the code example here. 👍

(+ 1 2)
```

### 3. Create a function that takes and argument and calls function defined in (2) with it

> Remember, you can invoke a function you defined the same way you invoke other functions- by putting its name as the first argument of a Clojure form and letting the Clojure compile evalute it!

Reviewer 3: ```suggestion
Remember, you can invoke a function you defined the same way you invoke other functions- by putting its name as the first argument of a Clojure form and letting the Clojure compiler evaluate it.
```

> Remember, you can invoke a function you defined the same way you invoke other functions- by putting its name as the first argument of a Clojure form and letting the Clojure compile evalute it!

Reviewer 2: ```suggestion
Remember, you can invoke a function you defined the same way you invoke other functions- by putting its name as the first argument of a Clojure form and letting the Clojure compiler evaluate it!
```
